ext/gmp/gmp.c

```diff
@@ -551,10 +588,19 @@ ZEND_MINIT_FUNCTION(gmp)
 
 	register_gmp_symbols(module_number);
 
+	mp_get_memory_functions(&gmp_old_alloc, &gmp_old_realloc, &gmp_old_free);
+	mp_set_memory_functions(gmp_alloc, gmp_realloc, gmp_free);
```

Comment:
Maybe we should be nice and restore the old functions on MSHUTDOWN?

Reply:
I think that would be better yes.